### PR TITLE
TSWEB-955 Added tests for Healthcare page

### PR DESCRIPTION
**Pull request by Valerii-Kryvokhyzha**

### CODE REVIEW

**Review by DaniilVostrikov — APPROVED** [review]

File: ui_tests_playwright/automation/base/step/BaseDriverSteps.ts [hunk 1]
```diff
@@ -41,6 +42,33 @@ class BaseDriverSteps {
...
+		await carouselButtonNext.click({delay: 1000});
+
+		await expect(carouselButtonPrev).toHaveAttribute('data-disabled', 'false');
+		await expect(carouselButtonNext).toHaveAttribute('data-disabled', 'false');
+		await carouselButtonPrev.click({delay: 1000});
+
+		await expect(carouselButtonPrev).toHaveAttribute('data-disabled', 'true');
+		await expect(carouselButtonNext).toHaveAttribute('data-disabled', 'false');
+
+		const clickAmount = clicksCount ? clicksCount : allSectionTitles.length - 1;
```
Comment (VladFedash) [inline ui_tests_playwright/automation/base/step/BaseDriverSteps.ts:63]: I don't see a reason to create additional variable, you can use clicksCount instead and overwrite it if it needed
clicksCount = clicksCount ? clicksCount : allSectionTitles.length - 1

File: ui_tests_playwright/automation/preconditionsData/Links.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	static Twitter = 'https://twitter.com';
+	static GoodFirms = 'https://www.goodfirms.co/company/techstack-ltd';
+	static Clutch = 'https://clutch.co/profile/techstack';
+
+	static InstagramDesign = 'https://www.instagram.com/techstack.design/';
+	static TikTokDesign = 'https://www.tiktok.com/@techstack.design';
+
+	static BehanceDimaDityuk = 'https://www.behance.net/dimadityuk';
+
+	static nuget = 'https://www.nuget.org/profiles/VitaliiDolotov';
```
Comment (VladFedash) [inline ui_tests_playwright/automation/preconditionsData/Links.ts:16]: Why is it different from other items?

**Review by anastasiia-liashova — APPROVED** [review]

File: ui_tests_playwright/automation/test/ui/CareerSite/aboutUs/AboutUsBlock.test.ts [hunk 3]
```diff
@@ -1,46 +1,171 @@
...
+
+	await expect(engineeringCulturePrevButton).toHaveAttribute('data-disabled', 'true');
+	await expect(engineeringCultureNextButton).toHaveAttribute('data-disabled', 'false');
+
+	await engineeringCultureNextButton.click();
+
+	await expect(engineeringCulturePrevButton).toHaveAttribute('data-disabled', 'false');
+	await expect(engineeringCultureNextButton).toHaveAttribute('data-disabled', 'false');
+
+	await engineeringCultureNextButton.click({clickCount: 3});
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/CareerSite/aboutUs/AboutUsBlock.test.ts:128]: What does the number '3' mean?
Why did we choose this value?

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+test("Check redirect by Clutch button in 'Beats Screening Module by Techstack' container from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
+	const beatsScreeningModuleContainer = driver.getByTestId(Healthcare.BeatsScreeningModuleByTechstack);
+
+	await beatsScreeningModuleContainer.getByTestId(Button.Clutch).click();
+	const newPage = await driver.DriverContext.waitForEvent('page');
+	expect(newPage.url()).toContain(ClutchReviewLinks.AnonymousMedicalDevice);
+});
+
+test("Check redirect by 'Read the full Case Study button' in 'Beats Screening Module by Techstack' container from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts:33]: I think the button shouldn't be inside '' quotes, isn't it?

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+});
+
+test("Check redirect by Breadcrumbs in header from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
+	const info = driver.getByTestId(Healthcare.Info);
+	await info.getByTestId(Container.BreadcrumbsPrev).click();
+
+	await baseDriverSteps.checkUrl(UrlProvider.webSiteUrl());
+});
+
+test("Check redirect by Clutch button in 'Beats Screening Module by Techstack' container from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts:25]: Why did you add quotes below for the button name, but skipped here?

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+import UrlPath from '../../../../../providers/UrlPath';
+import UrlProvider from '../../../../../providers/UrlProvider';
+import {Environment} from '../../../../../providers/EnvProvider';
+import Container from '../../../../../identifiers/MainSite/Container';
+
+test.beforeEach(async () => {
+	await baseDriverSteps.createsNewBrowserAndGoToUrl(UrlProvider.urlBuilder(UrlPath.Healthcare));
+});
+
+test("Check redirect by Breadcrumbs in header from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts:18]: same as below 
What is the object Breadcrumbs? Is it link or button or smth other?

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+	for (const entries of linkMap.entries()) {
+		await mostRecentIndustryFactsContainer.getByTestId(entries[0]).first().click();
+		const newPage = await driver.DriverContext.waitForEvent('page');
+		expect(newPage.url()).toContain(entries[1]);
+		await newPage.close();
+	}
+});
+
+test("Check carousel sections, arrows and CTA button in 'How We Operate' container from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts:64]: same for button name

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+			'https://www.statista.com/outlook/dmo/ecommerce/beauty-health-personal-household-care/health-care/worldwide',
+		],
+		[MainSiteButton.Pwc, 'https://www.pwc.com/us/en/industries/health-industries/library/healthcare-trends.html'],
+		[
+			MainSiteButton.McKinseyHealthcare,
+			'https://www.mckinsey.com/industries/healthcare/our-insights/what-to-expect-in-us-healthcare-in-2023-and-beyond#/',
+		],
+	]);
+
+	for (const entries of linkMap.entries()) {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts:56]: Should 'entries' be const or let? 
I saw in other files a similar foreach with let. Choose o ne right option

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+	]);
+
+	await baseDriverSteps.checkCarouselArrowsClick(howWeOperateContainer);
+
+	await expect(howWeOperateContainer.getByTestId(MainSiteButton.ScheduleAMeeting)).toBeVisible();
+});
+
+test("Check redirects by arrows in 'Core Practices' container from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
+	const corePracticesContainer = driver.getByTestId(Healthcare.CorePractices);
+
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts:91]: remove lb

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+		[containerSection.nth(0).getByTestId(Container.Arrow), UrlProvider.urlBuilder(UrlPath.CustomDev)],
+		[containerSection.nth(1).getByTestId(Container.Arrow), UrlProvider.urlBuilder(UrlPath.CloudAndDev)],
+		[containerSection.nth(2).getByTestId(Container.Arrow), UrlProvider.urlBuilder(UrlPath.BigData)],
+		[containerSection.nth(3).getByTestId(Container.Arrow), UrlProvider.urlBuilder(UrlPath.InternetOfThings)],
+		[containerSection.nth(4).getByTestId(Container.Arrow), UrlProvider.urlBuilder(UrlPath.AiMl)],
+		[containerSection.nth(5).getByTestId(Container.Arrow), UrlProvider.urlBuilder(UrlPath.MobileDev)],
+		[containerSection.nth(6).getByTestId(Container.Arrow), UrlProvider.urlBuilder(UrlPath.UiUxDesign)],
+	]);
+
+	for (const [arrow, url] of arrowUrlMap) {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareActions.test.ts:103]: s

File: ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareContent.test.ts [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+		'Health\nMonitoring Software',
+		'UI Kit',
+	];
+
+	expect(allSectionTitles.sort()).toEqual(testData.sort());
+
+	await expect(ourExpertiseContainer.getByTestId(MainSiteButton.ScheduleAMeetingNow)).toBeVisible();
+});
+
+test("Check section titles, image and CTA button in 'Beats Screening Module by Techstack' container from the 'Healthcare' block @Regression @Healthcare @TSWEB-955", async () => {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/Healthcare/HealthcareContent.test.ts:114]: s

File: ui_tests_playwright/automation/test/ui/generalComponents/generalContactUs/ContactUs.test.ts [hunk 12]
```diff
@@ -4,22 +4,30 @@ import {baseDriverSteps} from '../../../../base/step/BaseDriverSteps';
...
+import {serviceUrl, companyUrl, industryUrl} from '../../../../preconditionsData/UrlPreconditions';
 import {CompanyEnum} from '../../../../enum/CompanyEnum';
 
 test.beforeEach(async () => {
 	await baseDriverSteps.createsNewBrowser();
 });
 
-test.skip("Check 'Contact Us' button from 'Services' section @Regression @ContactUs @TSWEB-532", async () => {
+test("Check 'Contact Us' button from 'Industries' section @Regression @ContactUs", async () => {
+	for (const url of Object.values(industryUrl)) {
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/generalComponents/generalContactUs/ContactUs.test.ts:15]: s

**Review by VladFedash — CHANGES_REQUESTED** [review]

**Review by UncleDanya — APPROVED** [review]

**Review by VladFedash — APPROVED** [review]

**Review by DaniilVostrikov — APPROVED** [review]

**Pull request merged**